src/components/ErrorBoundary/__tests__/ErrorBoundary.test.tsx

```diff
@@ -0,0 +1,32 @@
+import { useState } from 'react';
+import { ErrorBoundary } from '../ErrorBoundary';
+import { render, fireEvent, screen, waitFor } from '@testing-library/react';
+
+test('renders ErrorBoundary and fire an error', async () => {
+  function BuggyCounter() {
+    const [counter, setCounter] = useState<any>(0);
+
+    function handleClick() {
+      setCounter(counter + 1);
+    }
+
+    if (counter === 1) {
+      // Simulate a JS error
+      throw new Error('SIMULATE INTENTIONAL CRASH CASES -- This is fine :)');
+    }
+
+    return (
+      <h1 id="error-button" onClick={handleClick}>
```

Comment:
I would change `<h1>` to `<button>` and then modify also slightly test case to click button with the id `error-button`

Reply:
React testing-library provide a property named data-testid, I think this one would be the best for this situation. https://testing-library.com/docs/queries/bytestid